lms/services/user.py

```diff
@@ -17,10 +17,17 @@ class UserService:
     At the moment this is purely used for recording/reporting purposes.
     """
 
-    def __init__(self, application_instance_service, db, h_authority: str):
+    def __init__(self, application_instance_service, db, h_authority: str, request):
         self._application_instance_service = application_instance_service
         self._db = db
         self._h_authority = h_authority
+        self._request = request
```

Comment:
Including here `request.lti_user` instead of the whole request resulted in a circular dependency y between the two `config.add_request_method` added properties, lti_user and user.